.github/scripts/check-for-go-mod-changes.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+set -ue
+
+for DIRECTORY in . ./pkg/apis ./pkg/client; do
+    cd "$DIRECTORY"
+    go mod tidy
+    go mod vendor
```

Comment:
Don't believe we need to run vendor. 
Could you make whole thing more similar to our validate script, that has other important checks, specifically go-mod-verify, and the pkg/apis go.mod check: https://github.com/rancher/rancher/blob/release/v2.7/scripts/validate#L20-L23

Or maybe we could run that script directly ?